packages/lodestar/src/eth1/impl/eth1Vote.ts

```diff
@@ -0,0 +1,67 @@
+import {IEth1Notifier} from "../interface";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {BeaconState, Eth1Data, number64} from "@chainsafe/eth2.0-types";
+import {intSqrt} from "@chainsafe/eth2.0-utils";
+import {arrayIntersection, mostFrequent, sszEqualPredicate} from "../../util/objects";
+import {equals} from "@chainsafe/ssz";
+import {Block} from "ethers/providers";
+
+export async function getEth1Vote(
+  this: IEth1Notifier, config: IBeaconConfig, state: BeaconState, previousEth1Distance: number64
+): Promise<Eth1Data> {
+  const eth1Head = await this.getHead();
+  const allEth1Data = await getEth1DataRange(
+    this.getEth1Data,
+    eth1Head,
+    config.params.ETH1_FOLLOW_DISTANCE,
+    previousEth1Distance
```

Comment:
seems like there's an opportunity to cache eth1datas from known distances? I guess this is what Adrian was mentioning in the Eth2 call the other day.

Reply:
probably, but I think we can open separate issue for that?

Reply:
Yeah, for sure, just mentioning